import_addressbase/import_csv.py

```diff
@@ -0,0 +1,180 @@
+                '_type': doc_type,
+                '_id': dpa.uprn,
+                '_source': doc,
+            }
+        elif dpa.change_type == 'U':
+            action_dict = {
+                '_op_type': 'update',
+                '_index': INDEX_NAME,
+                '_type': doc_type,
+                '_id': dpa.uprn,
+                'doc': doc,
+            }
+        elif dpa.change_type == 'D':
+            action_dict = {
+                '_op_type': 'delete',
+                '_index': INDEX_NAME,
+                '_type': doc_type,
+                '_id': dpa.uprn,
+            }
+        return action_dict
+
+    actions = [make_action(doc_type) for doc_type, _ in TYPE_TO_INDEX_MAPPING]
+    return actions
+
+
+def get_action_dicts(filename):
+    """A generator which yields elasticsearch action dicts for groups of records
+    with one DPA and zero or one BPLU
+    """
+    with open(filename, 'r') as csv_file:
+        data_reader = csv.reader(csv_file)
+
+        entry_datetime = None
+
+        for _, group in groupby(data_reader, itemgetter(UPRN)):
+            rows = list(group)
+            if (len(rows) == 1 and
+                    int(rows[0][RECORD_IDENTIFIER]) == HEADER_ID):
+                header = Header(*rows[0])
+                # we use 'date_time_no_millis' format: yyyy-MM-dd’T'HH:mm:ssZZ
+                # we assume UTC (+00) as the spec doesn't specify a timezone
+                entry_datetime = '{}T{}+00'.format(header.entry_date,
+                                                   header.time_stamp)
+
+                continue
+
+            blpu_list = []
+            dpa_list = []
+            # create namedtuples from each line
+            for row in rows:
+                rec_type = int(row[RECORD_IDENTIFIER])
+                # create a record using the values in the row
+                if rec_type == BLPU_ID:
+                    blpu_list += [BLPU(*row)]
+                elif rec_type == DPA_ID:
+                    dpa_list += [DPA(*row)]
+
+            # we must have one DPA and zero or one BPLU
+            if len(dpa_list) == 1 and len(blpu_list) in [0, 1]:
+                dpa = dpa_list[0]
+                position = None
+                if len(blpu_list) == 1:
+                    blpu = blpu_list[0]
+                    position = {
+                        'x': float(blpu.x_coordinate),
+                        'y': float(blpu.y_coordinate)
+                    }
+                action_dicts = make_es_actions(dpa, position, entry_datetime)
+
+                # TODO: remove debug print statement
+                from pprint import pprint
+                pprint(action_dicts, width=1)
+
+                for action_dict in action_dicts:
+                    yield action_dict
+
+
+def import_csv(filename, nodes):
+    client = Elasticsearch(nodes)
+    # create index if it doesn't exist
+    client.index(index=INDEX_NAME, doc_type=TYPE_TO_INDEX_MAPPING[0][0],
```

Comment:
Perhaps it would be best to refer to `property` type directly, using a constant, for example (`doc_type=PROPERTY_TYPE`).